src/Tvheadend.cpp

```diff
@@ -1723,54 +1728,128 @@ PVR_ERROR CTvheadend::OnSystemWake()
  * VFS
  * *************************************************************************/
 
-bool CTvheadend::OpenRecordedStream(const kodi::addon::PVRRecording& rec)
+bool CTvheadend::OpenRecordedStream(const kodi::addon::PVRRecording& recording, int64_t& streamId)
 {
   if (!m_asyncState.WaitForState(ASYNC_EPG))
+    return PVR_ERROR_SERVER_TIMEOUT;
+
+  const auto vfs{std::make_shared<HTSPVFS>(m_settings, *m_conn)};
+  if (!vfs->Open(recording))
     return false;
 
-  bool ret = m_vfs->Open(rec);
+  streamId = vfs->GetFileId();
```

Comment:
Since the map key (`streamId`) is deduced from the map value (`vfs`), would it be simpler to store a `std::vector<std::shared_ptr<tvheadend::HTSPVFS>>` and find the correct instance with a lambda?

Reply:
What do you mean exactly? Not sure I understand, sorry.

Reply:
Now I got it. I added the streamId later to HTSPVFS... Anyway, wouldn't this have a negative performance impact (linear seek)?

Reply:
In theory it will probably be slower, but for the most part there will only be one item in the vector/map so it shouldn't matter much (if I understand correctly, the only time there will be multiple streams read is when Kodi is generating thumbnails for a list of recordings).

If changing to a vector doesn't simplify the code then I don't think it's worth changing what you have now.

Reply:
Fair enough. Anyway, i think I will go with what I have now.